### DISCUSSION TITLE
Working with companion type for a resolved TypeRepr or Symbol

### DISCUSSION THREAD

**Discussion by cchantep** [scala/scala3#17825] [Feature Requests]
In Scala 2, it was possible to:

- from a `t: Type` do `val companionTpe = t.companion`
- from a `s: Symbol` do `val companionTpe = s.companion.typeSignature`

And then for example to work with the `unapply` resolved through the companion.

```scala
companionTpe.decl(TermName("unapply"))
```

There is no such equivalent in Scala 3.

A function a `companion` could be implemented as below on [`TypeReprMethods`](https://github.com/lampepfl/dotty/blob/master/compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala#L1705).

```scala
def companion: TypeRepr =
  TypeRepr.typeConstructorOf(Class.forName(self.typeSymbol.fullName + '$'))
```

**Comment by LPTK** [comment]
Seems like a good addition to me. Why don't you make a PR directly on the Dotty repo?

(I'm not affiliated with the compiler team.)

**Comment by cchantep** [comment]
I would be able to do so iff how the tests are executed was clear/documented.

**Reply by ckipp01** [reply]
I know this was long ago, but please do check out the docs [here](https://dotty.epfl.ch/docs/contributing/testing.html). If you feel like something isn't clear or is missing, report back.

**Comment by LPTK** [comment]
Running the SBT `test` command should be enough. This is likely not going to introduce a regression. PRs are automatically tested anyway.

**Comment by cchantep** [comment]
No it's not, and it lacks a proper documentation.
